core/src/core/transaction.rs

```diff
@@ -306,6 +306,34 @@ impl TxKernel {
 		Ok(())
 	}
 
+	/// Batch verification.
+	pub fn batch_verify(tx_kernels: &Vec<TxKernel>) -> Result<(), Error> {
+		let len = tx_kernels.len();
+		let mut sigs: Vec<secp::Signature> = Vec::with_capacity(len);
+		let mut pubkeys: Vec<secp::key::PublicKey> = Vec::with_capacity(len);
+		let mut msgs: Vec<secp::Message> = Vec::with_capacity(len);
+
+		let secp = static_secp_instance();
+		let secp = secp.lock();
+
+		for tx_kernel in tx_kernels {
+			if tx_kernel.is_coinbase() && tx_kernel.fee != 0
```

Comment:
Can we split the fee and lock_height verification out from this batch signature verification?
There are some pending changes in #2859 that rework this and these don't need to be batched up.
We can just do 2 passes over the kernels -
* verify fee and lock_height
* then batch verify the signatures

Reply:
Sure, I can split it out and will rework it.
BTW, when do you plan to complete #2859?

Reply:
will try the splitting today and request a new review on this.